### PR TITLE
GeoPackage - Field domain added to Browser

### LINKED ISSUE #7365: Add field domain management capabilities to browser (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#47025
Author: @nyalldawson
QGIS version: 3.26 (Feature)

**Add field domain management capabilities to browser**

### PR Description:
This PR adds API to manage field domains to database connections, and exposes this functionality through the QGIS browser panel.

Some notes:

- This functionality is currently exposed only for Geopackages, but there is support in GDAL for field domain handling for other database types (ESRI FileGeodatabase, ESRI Personal Geodatabases) which could potentially be exposed in future. We could also potentially expose this functionality for the native database providers in future too (the API is designed to easily permit this)
- The functionality requires GDAL 3.5+
- QGIS already checks field domains when loading GPKG layers in order to setup nice defaults for editor widgets, but there's only partial support there (e.g. no support for glob domains in the QGIS editor widgets, or non-inclusive range domains). This functionality isn't changed by this PR.
- The new "Field Domains" item is only shown when expanding out GeoPackages which have some existing field domains (to avoid unwanted clutter)

TODO:

- [x] Expose ability to set the field domain for GPKG table fields
- [x] Expose ability to create new field domains
- [ ] Expose ability to delete and update field domain definitions (when supported by GDAL)

Demo showing display of existing domain properties:

Peek 2022-01-27 15-14

### Commits tagged with [need-docs] or [FEATURE]
"[feature] Add browser properties page for field and fields items\n\nIn particular, this allows users to easily see what field domain\nmay be attached to particular fields" "[feature] Show \"Set Field Domain\" action when right clicking\na field in the browser for a Geopackage table\n\nAllows users to set the field domain associated with that field"

### PR DESCRIPTION
**Pull request by selmaVH1**
fix #7365 
To be honest, I'm not sure if this is enough to close the issue, but I couldn't find where else this information would be relevant.

- [x] Backport to LTR documentation is requested

### CODE REVIEW

**Comment by DelazJ** [comment]
Yeah, I agree with you. And I'm afraid the current structure of the docs doesn't provide obvious place for these details (this table with checkboxes doesn't help with details).
Maybe could you howwver mention in "vector layer loading" chapter that fields with "fields domain" are automatically recognized by QGIS and assigned the appropriate (map value?) widget, and/or add such a note in the widgets edit section? There are some instructive details in the issue report discussion.

**Comment by selmaVH1** [comment]
> Maybe could you howwver mention in "vector layer loading" chapter that fields with "fields domain" are automatically recognized by QGIS and assigned the appropriate (map value?) widget, and/or add such a note in the widgets edit section? There are some instructive details in the issue report discussion.

@DelazJ Thanks for pointing me to the right place, I added notes to both.

File: docs/user_manual/managing_data_source/opening_data.rst [hunk 1]
```diff
@@ -421,6 +421,19 @@ To load a layer from a file:
...
 .. index:: US Census Bureau
 
+.. note::
+
+ When loading vector layers into QGIS, fields with |sourceFields| :guilabel:`Field Domains`
+ (such as those defined in a GeoPackage or ESRI File Geodatabase) are automatically
+ detected. These domains are database-level constraints, meaning they are enforced
+ by the database itself and apply across different applications, not just QGIS.
+ This differs from the QGIS-only **Value Maps** widget, which is saved within QGIS project styles
+ and is not enforced outside QGIS.
```
Comment (DelazJ) [inline docs/user_manual/managing_data_source/opening_data.rst:431]: actually, you have:
* "coded values" automatically detected by "value maps" widget
* "range domains" automatically detected by "range" widget
* and "glob" ... I couldn't find the syntax to better experiment if it is recognized by something.

```suggestion
 This differs from the QGIS-only widgets, which are saved within QGIS project styles
 and are not enforced outside QGIS.
```
or just remove to focus on the feature, at the db level?

**Review by DelazJ — COMMENTED** [review]
I've played a bit with the feature and have some more info

**Pull request merged**